fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/settings/SettingsFragment.java

```diff
@@ -0,0 +1,75 @@
+package com.github.dedis.popstellar.ui.settings;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+
+import com.github.dedis.popstellar.databinding.SettingsFragmentBinding;
+import com.github.dedis.popstellar.repository.remote.LAORequestFactory;
+import com.github.dedis.popstellar.ui.home.HomeActivity;
+
+public class SettingsFragment extends Fragment {
+
+  private final String TAG = SettingsFragment.class.getSimpleName();
+
+  private SettingsFragmentBinding mSettingsFragBinding;
+  private SettingsViewModel mSettingsViewModel;
+
+  public static SettingsFragment newInstance() {
+    return new SettingsFragment();
+  }
+
+  @Override
+  public View onCreateView(
+      @NonNull LayoutInflater inflater,
+      @Nullable ViewGroup container,
+      @Nullable Bundle savedInstanceState) {
+    mSettingsFragBinding = SettingsFragmentBinding.inflate(inflater, container, false);
+
+    mSettingsViewModel = SettingsActivity.obtainViewModel(requireActivity());
+
+    mSettingsFragBinding.setViewmodel(mSettingsViewModel);
+    mSettingsFragBinding.setLifecycleOwner(getViewLifecycleOwner());
+
+    return mSettingsFragBinding.getRoot();
+  }
+
+  @Override
+  public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+    super.onViewCreated(view, savedInstanceState);
+
+    setupApplyButton();
+    mSettingsViewModel.setServerUrl(LAORequestFactory.getUrl());
+    mSettingsViewModel.setTempServerUrl(LAORequestFactory.getUrl());
+
+    // Subscribe to "apply changes" event
+    mSettingsViewModel
+        .getApplyChangesEvent()
+        .observe(
+            getViewLifecycleOwner(),
+            booleanEvent -> {
+              Boolean event = booleanEvent.getContentIfNotHandled();
+              if (event != null) {
+                applyChanges();
+              }
+            });
+  }
+
+  private void setupApplyButton() {
+    mSettingsFragBinding.buttonApply.setOnClickListener(v -> mSettingsViewModel.applyChanges());
+  }
+
+  private void applyChanges() {
+    LAORequestFactory.setUrl(mSettingsFragBinding.entryBoxServerUrl.getText().toString());
```

Comment:
And here, as the function `setUrl`cannot be accessed statically, we need to get the instance from the Injection class.

```suggestion
    Injection.provideRequestFactory().setUrl(mSettingsFragBinding.entryBoxServerUrl.getText().toString());
```

Reply:
oh ok it works that way, thank you for pointing out !